docs/source/use_cases/run_picard_and_gatk/index.rst

```diff
@@ -17,20 +17,178 @@ __ RenderedVersion_
+We benchmarked **x11** speed improvements with GRPC compared to REST, giving **~12,000 reads/second**.
+
+The tests were done on `Platinum Genomes NA12877_S1.bam dataset <https://pantheon.corp.google.com/storage/browser/genomics-public-data/platinum-genomes/bam/?_ga=1.197206447.160385476.1431305548>`_, please see the `detailed writeup of the test procedure and results <https://docs.google.com/document/d/1Br7RMSbAChNpG6pi2teujf-YthczF-rAM1afSgDoCgQ/edit#>`_ if you want to repeat the test.
+
+We therefore recommend running GRPC variants of command line.
+
+Other Picard tools you can run
+------------------------------
+You can run MarkDuplicates or MarkDuplicatesWithMateCigar tools like this:
+
+.. code-block:: shell
+
+  java \
+    -Xbootclasspath/p:../gatk-tools-java/lib/alpn-boot-8.1.3.v20150130.jar \
+    -Dga4gh.using_grpc=true \
+    -jar dist/picard.jar MarkDuplicates \
+    INPUT=https://www.googleapis.com/genomics/v1beta2/readgroupsets/CK256frpGBD44IWHwLP22R4/ \
+    OUTPUT=output.bam \
+    METRICS_FILE=output.metrics \
+    GA4GH_CLIENT_SECRETS=../client_secrets.json
+
+Figuring out a url for your dataset
+-----------------------------------
+In the examples above we have been using urls of the form https://www.googleapis.com/genomics/v1beta2/readgroupsets/XXX where XXX is the id of the readset. 
+
+How do you find an ID of the readset from the `Discover Published Data`_ set or from your own project ? 
+
+We will do it step by step using the command line API client. 
+
+* Lets say we want to use `Platinum Genomes NA12877_S1.bam readgroupset <https://pantheon.corp.google.com/storage/browser/genomics-public-data/platinum-genomes/bam/?_ga=1.197206447.160385476.1431305548>`_ from `genomics-public-data <https://cloud.google.com/genomics/data/1000-genomes?hl=en>`_ project.
+
+* The `documentation <https://cloud.google.com/genomics/data/1000-genomes?hl=en>`_ page states that the dataset id for this set of files is **10473108253681171589**.
+
+* To list readgroupsets under this dataset:
+
+.. code-block:: shell
+
+     $ cloud alpha genomics readgroupsets list 10473108253681171589 --limit 10
+     ID                      NAME     REFERENCE_SET_ID
+     CMvnhpKTFhDq9e2Yy9G-Bg  HG02573  EOSt9JOVhp3jkwE
+     CMvnhpKTFhCEmf_d_o_JCQ  HG03894  EOSt9JOVhp3jkwE
+     ...
+
+* Note the **NAME** column - it will correspond to the file name. The **ID** column is the ID of the readgroupset we are looking for.
+
+
+Now lets suppose we are not looking for one of the readgroupsets form the genomics public data but instead want to use one from our own project.
+In this case we need to figure out the *dataset id* for our files first, before we can use "readgroupsets list" command to list the individual readgroupsets.
+
+* Lets say we want to figure out which dataset ids are present under `genomics test data <https://pantheon.corp.google.com/project/genomics-test-data/storage/browser>`_ project.
```

Comment:
https://console.developers.google.com instead of https://pantheon.corp.google.com